### PR TITLE
Created many to many association for Users and Cities

### PR DESCRIPTION
**Pull request by peregrinebalas**
Added:
Models
- UserCity
- City
favoritesRouter
- post
- get

Updated:
forecastRouter
- get (findOrCreate City)

### CODE REVIEW

File: models/user.js [hunk 1]
```diff
@@ -11,9 +11,24 @@ module.exports = (sequelize, DataTypes) => {
     },
     password: DataTypes.STRING,
     apiKey: DataTypes.STRING
-  }, {});
-  User.associate = function(models) {
-    // associations can be defined here
+  }, {
+    instanceMethods: {
+      getCities: function() {
```
Comment (peregrinebalas) [inline models/user.js:17]: @plapicola this is the instance method I mentioned to call all associated cities, that's returning a promise in the Router. I've just used pry to test what the return is. The request is making the right query to the db but not resolving. Any ideas for fixes?

Comment (allieIsOkay) [reply models/user.js:17]: If I had to guess, the issue is that user.getCities() is returning another promise, so when the send query calls it, it'll send the response without waiting for the content. Breaking it out in to it's own then block before the send might resolve the issue. I'd try something like this: 

``` javascript
User.findOne({
    where: {
      apiKey: req.body.apiKey
    }
  }).then(user => {
    return user.getCities()
  }).then(cities => {
    eval(pry.it)
    res.setHeader("Content-Type", "application/json")
    res.status(200).send(cities)
  })
```

I tried this after cloning the repo to my local machine, and it at least returned an empty array, but without adding the full functionality for creating city records in the database I can't say for certain.

Comment (allieIsOkay) [reply models/user.js:17]: Another option that might reduce the number of database queries is to include the city records on the initial user find, then just return the inner part of the resulting object. Doing it this way you can find your user by the API key and get their locations in a single query.

Comment (peregrinebalas) [reply models/user.js:17]: I think that did the trick! Thank you! I'm gonna get adding favorites up and running to make sure but this is a good start!

**Pull request merged**